### PR TITLE
8343981: Remove usage of security manager from Thread and related classes

### PR DESCRIPTION
**Pull request by AlanBateman**
Removes the SecurityManager usage from Thread + friends.

In Thread, the getContextClassLoader method is no longer caller-sensitive method.

JavaLangAccess.newThreadWithAcc is removed and jdk.internal.access is no longer exported to java.naming. The usage of newThreadWithAcc is removed from com.sun.jndi.ldap.VersionHelper. There will be further work on the java.naming module to remove usage of SM, the change here is specific to the usage of ewThreadWithAcc.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8343981](https://bugs.openjdk.org/browse/JDK-8343981): Remove usage of security manager from Thread and related classes (**Enhancement** - P4)

### Reviewers
 * [Roger Riggs](https://openjdk.org/census#rriggs) (@RogerRiggs - **Reviewer**)
 * [Yudi Zheng](https://openjdk.org/census#yzheng) (@mur47x111 - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/22035/head:pull/22035` \
`$ git checkout pull/22035`

Update a local copy of the PR: \
`$ git checkout pull/22035` \
`$ git pull https://git.openjdk.org/jdk.git pull/22035/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 22035`

View PR using the GUI difftool: \
`$ git pr show -t 22035`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/22035.diff">https://git.openjdk.org/jdk/pull/22035.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/22035#issuecomment-2471046262)
</details>

### CODE REVIEW

**Comment by dfuch** [comment]
The proposed changes look reasonable to me. Good simplification of the code. I had some uncertainty about the JNDI changes but I couldn't find any place in JNDI where a Subject would be extracted from the ACC associated with the created thread so I'd say that looks fine.

**Comment by AlanBateman** [comment]
> The proposed changes look reasonable to me. Good simplification of the code. I had some uncertainty about the JNDI changes but I couldn't find any place in JNDI where a Subject would be extracted from the ACC associated with the created thread so I'd say that looks fine.

ACC is dead and Subject is re-implemented on ScopedValue now so I think we are okay. There will be many other places where we will also drop the capture and use of the ACC.

File: src/java.naming/share/classes/com/sun/jndi/ldap/VersionHelper.java [hunk 1]
```diff
@@ -109,14 +107,8 @@ Class<?> loadClass(String className) throws ClassNotFoundException {
...
 
-    @SuppressWarnings("removal")
     Thread createThread(Runnable r) {
-        AccessControlContext acc = AccessController.getContext();
-        // 4290486: doPrivileged is needed to create a thread in
-        // an environment that restricts "modifyThreadGroup".
-        PrivilegedAction<Thread> act =
-                () -> SharedSecrets.getJavaLangAccess().newThreadWithAcc(r, acc);
-        return AccessController.doPrivileged(act);
+        return new Thread(r);
```
Comment (RogerRiggs) [inline src/java.naming/share/classes/com/sun/jndi/ldap/VersionHelper.java:111]: Can the refactoring go back to the callers of createThread() to just create threads themselves.
And remove this trivial method.

Comment (AlanBateman) [reply src/java.naming/share/classes/com/sun/jndi/ldap/VersionHelper.java:111]: Yes, but in the PR that does the cleanup of java.naming module.

**Review by RogerRiggs — APPROVED** [review]
LGTM

**Comment by mur47x111** [comment]
Please hold until I have the Graal changes ready

**Review by mur47x111 — APPROVED** [review]

**Pull request closed without merging**